CDAP-6321 Master Service Bind Port Not Configurable

bennythomps opened the pull request:
Added ability to configure port and bind addresses for master services (via CDAP conf).

ajainarayanan committed:
Fixes number widget to get proper max and min values from widget json & convert number 0 to string \'0\'

ajainarayanan committed:
Fixes min & max attributes to number widget in hydrator++ to be strings

ajainarayanan committed:
Fixes timeline js to be added as es6Directives in Gulpfile.js

ajainarayanan committed:
Fixes styling of modal (hide scroll outside the modal tabs) in hydrator++

dereklwood committed:
add script to obtain Ambari credentials, configure pageblob support, and restart required services

dereklwood committed:
remove ambari connectivity, update cdap/chef versions

dereklwood committed:
need to specify version for cli install, otherwise could end up with mixed versions

dereklwood committed:
ntp now provided by HDinsight

dereklwood committed:
configure tez for cdap

dereklwood committed:
cdap solution template for v2 marketplace

albertshau committed:
CDAP-6375 refactor hydrator spark

Pulling Spark related Hydrator logic into its own module in
preparation for the new artifact that will run pipelines using
Spark Streaming.

As part of this refactor, a single Spark program can now be used
to run an entire pipeline, which removes the need for running the
planner logic and local connector datasets used by the mapreduce
engine.

dereklwood committed:
lots of cleanup per PR feedback

dereklwood committed:
more appropriately named function

dereklwood committed:
better context messages, cleanup

PGuimarais committed:
Log table retrieves and populates from endpoint

sagarkapare committed:
CDAP-6502 API changes and corresponding implementation int the DataPipelineApp for supporting Action plugin.

PGuimarais committed:
Adds timeline polling

elfenheart committed:
complete complex schema feature parity
update schema with implicit schema
add stream special case
add dataset selected event listener
remove clearDOM setting on schema clear event
fix error blink when switching to complex schema types
grouping schema types for simple and complex
handle stream import/export schema
cleanup commented out code
remove comment from dataset selected event
add spec-version 1.3
add error message when trying to export empty schema
modify compile to use es6 literals

PGuimarais committed:
Enables basic rendering of timeline data

PGuimarais committed:
Refactors link function

PGuimarais committed:
Refactors code and corrects timeline bug

PGuimarais committed:
Timeline scales with incoming logs

PGuimarais committed:
Restructures logviewer for single context

PGuimarais committed:
LogViewer and Timeline now follow the same time

PGuimarais committed:
Resolves re-painting bug

PGuimarais committed:
resolves bug for killed services

PGuimarais committed:
Makes directive reusable

PGuimarais committed:
Removed commented code

PGuimarais committed:
Addresses comments

PGuimarais committed:
Addresses comments

PGuimarais committed:
Addresses comments

ajainarayanan committed:
Fixes the case for checking 0 as its a falsy condition

PGuimarais committed:
Remedies inconsistent rendering of tick lines

PGuimarais committed:
Fixes expand-all functionality

dereklwood committed:
consistent indentation

PGuimarais committed:
Adjusts table height to viewport and adds slider pin

PGuimarais committed:
Adjust positioning of pin

PGuimarais committed:
Adds ellipsis to truncated message text

PGuimarais committed:
Adds sliderHandle image

dereklwood committed:
cleanly restart services by confirming all components stopped before starting

dereklwood committed:
cleanup

dereklwood committed:
compare as integers per PR feedback

chtyim committed:
(CDAP-6168) Includes all program options during program execution

- Preserves all program options
  - Needed so that all artifact information and runtime information
    can easily accessible
  - Serialize the program options for MapReduce and Spark
- Expose the artifact information from AbstractContext
- Refactor the AbstractContext hierarchy
  - Every program context now extends from AbstractContext
- Cleanup AbstractContext sub-classes to remove code duplication and
  proper common implmentation
- Simplify sub-class implementation
- Unify RunId construction from ProgramOptions

dereklwood committed:
fix indentation

PGuimarais committed:
Adds basic infinite scrolling capabilities

PGuimarais committed:
Adds bootstrap well for loading data

dereklwood committed:
fix json syntax in generated attributes for cdap-cli install

PGuimarais committed:
Formats table cells

PGuimarais committed:
Adds tooltip to source column

Commit:
[CDAP-6358] Added MacroParser and PluginInstantiator macro support

PGuimarais committed:
Adds loading icon to bottom of table

PGuimarais committed:
Adds line number to tooltip

gsps1 committed:
CDAP-6357 porting logical start time macro from hydrator-plugins repo

PGuimarais committed:
Infinite scroll now receives sequential data

ajainarayanan committed:
Adds error template to the top panel to sift through multiple errors

PGuimarais committed:
rerenders on start time change

ajainarayanan committed:
Adds alertOnValium to accept a templateUrl to be replaced in the alert message

ajainarayanan committed:
Fixes config store & console store to use appropriate addMessage API

ajainarayanan committed:
Fixes console store & top panel controllers to use the new addMessage API from console store

ajainarayanan committed:
Minor styling + template changes to top panel

ajainarayanan committed:
Fixes styling of canvas to hold just the enough height between top panel and the footer

ajainarayanan committed:
Cleansup un-used files in hydrator bottom panel in studio view

ajainarayanan committed:
Fixes pipeline non detail runs store & console stores to have listen & unsubscribe functions

ajainarayanan committed:
Removes un-used template files from the bottom panel in hydrator studio view

ajainarayanan committed:
Minor styling fixes for post run action + removes bottom panel from hydrator studio routes

dereklwood committed:
quote credentials for curl, in case of special chars in the password, due to eval

dereklwood committed:
find and query host_components to determine shutdown status

dereklwood committed:
cleanup

dereklwood committed:
initial createUiDef, restricting it to the latest HBase cluster version

dereklwood committed:
condense json formatting to 1-line where appropriate

ajainarayanan committed:
Adds my-pipeline-summar directive for top panel in hydrator

ajainarayanan committed:
Removes history controller from hydrator detailed view

ajainarayanan committed:
Removes history tab from bottom panel & adds my-pipeline-summary to top panel

ajainarayanan committed:
Adds history + summary cards to top panel

ajainarayanan committed:
Moves polling/requests as part of the pipeline summary to the respective directives

ajainarayanan committed:
Removes un-used templates in hydratorplusplus

ajainarayanan committed:
Fixes hydratorplusplus top panel controller based on separating pipeline-summary directive

ajainarayanan committed:
Minor conversion from if condition to ternary operator

dereklwood committed:
rename to required mainTemplate.json

dereklwood committed:
convert to 2-space indentation

Commit:
Fix cdap-default.xml and broken build.

Commit:
CDAP-6321 added ability to configure bind host and port for master services

ajainarayanan committed:
Adds pipelin export modal for viewing and exporting published pipelines

ajainarayanan committed:
Fixes published pipeline to use pipeline-modal

ajainarayanan committed:
Uses pipeline-export-modal for hydrator studio view to view and export pipeline config

ajainarayanan committed:
Removes configuration & metrics controller & templates from bottom panel in published view

astewes committed:
lowers min number of datasets from 3 to 1

astewes committed:
changes initial background color

astewes committed:
highlights Search tab when in usage state

anew commented:
The Travis build failed due to checkstyle errors. If you build the project using maven, it will run checkstyle. You can due a quick build using mvn package -DskipTests. If you are using IntelliJ, you can also download the settings that are compatible with our checkstyle rules from the wiki https://wiki.cask.co/display/CE/Coding+Standard and import them. IDEA will then automatically do most things right for you, such as ordering/optimizing imports, etc.

anew commented on cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/AppFabricServiceRuntimeModule.java:
```diff
@@ -344,8 +344,8 @@ protected void configure() {
     public InetAddress providesHostname(CConfiguration cConf) {
-      return Networks.resolve(cConf.get(Constants.AppFabric.SERVER_ADDRESS),
-                              new InetSocketAddress("localhost", 0).getAddress());
+      String address = cConf.get(Constants.AppFabric.SERVER_ADDRESS);
```
Did you mean to get the port number from the cConf, too? Otherwise this change is only rearranging the existing code...

bennythomps replied:
Agreed, this is not changing functionality, simply to avoid inlining the usage of the address. I can revert if the preference is to keep the resolution inline.

anew commented on cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/AppFabricServer.java:
```diff
@@ -49,11 +49,11 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import javax.annotation.Nullable;
```
this will probably break checkstyle, because it reorders the imports.

anew commented on cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/AppFabricServer.java:
```diff
@@ -147,6 +147,7 @@ protected void startUp() throws Exception {
     // Run http service on random port
     httpService = new CommonNettyHttpServiceBuilder(configuration)
       .setHost(hostname.getCanonicalHostName())
+      .setPort(configuration.getInt(Constants.AppFabric.SERVER_PORT))
```
this will throw a NullPointerException if the configuration does not contain Constants.AppFabric.SERVER_PORT. A default value of 0 needs to be added to cdap-common/src/main/resources/cdap-default.xml.

And the same for all the other newly introduced port configs.

anew commented on cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/StreamHttpService.java:
```diff
@@ -63,6 +63,7 @@ public StreamHttpService(CConfiguration cConf, DiscoveryService discoveryService
       .setHandlerHooks(ImmutableList.of(new MetricsReporterHook(metricsCollectionService,
                                                                 Constants.Stream.STREAM_HANDLER)))
       .setHost(cConf.get(Constants.Stream.ADDRESS))
+      .setPort(cConf.getInt(Constants.Stream.PORT))
```
The stream service does not run in the master node, but in a Yarn container. I am not sure whether it is a good idea to hardcode a port for services that run in containers, because there is no guarantee that the port is not taken by a service that runs in a different container, so this would randomly introduce failures at startup.

anew replied:
In fact, AppFabric and DatasetService are the only services that run in the master itself, all others run in containers. It may still make sense to configure these ports for the other services for a standalone CDAP.

bennythomps replied:
I left the configuration for all ports per your second comment (regarding standalone). Let me know if you'd prefer other services not configurable.

anew commented:
Thanks for sending this PR, @bennythomps, I made a pass through the changes and left a few comments, please take a look. -Andreas

albertshau committed:
CDAP-6507 fix hydrator phase splitting bug

Fixed bugs with phase splitting introduced when we added support
for multiple inputs. After that change, the previous algorithm
of splitting the dag on connector nodes could result in subdags
that contained multiple reduce nodes.

Changed the algorithm to ensure that each subdag contains at most
a single reduce node.

albertshau committed:
CDAP-6507 fix bug with dag splitting on simple case

Fix a bug where we try to add a dag without any nodes during
dag splitting.

Commit:
(CDAP-6516) Added a filter method to AuthorizationEnforcer, that can be used in any list/get API. This method filters the input set of entities to only return the ones that the user has access to. Also added a method to enforce multiple actions on the same entity.

Commit:
remove Log.Warn that printed out all configs in KafkaServerMain

ajainarayanan committed:
Adds run player dropdown. This will later have run time arguments that the user can enter to start/schedule a run in pipeline

gokulavasan committed:
Scaling RemoteSystemOperations by creating a new Guava service and start it in DatasetOpExecutorTwillRunnable

Should not be public-routable, used only through RemoteOpsClient

Add AuthModule to the DatasetOpTwillRunnable's injector since RemotePrivlege handler needs it

Adding RemoteSystemOp to system service status endpoint

Missed Inject annotation

Addressed comments

ajainarayanan committed:
Fixes minor styling in top panel for run player

Commit:
(CDAP-6337) Exclude asm dependency in deploy pom (#6077)

ajainarayanan committed:
Minor code lint + fetches average run time & next run time for an initial value and then starts polling for the same

Commit:
Moved default implementations of enforce and filter to AbstractAuthorizer

Commit:
(CDAP-6127) set default bind port to 0 for tx service (#6078)

sagarkapare committed:
CDAP-6528 Attempt Macro substituion only if the value is provided by user.

Commit:
CDAP-6154 Custom Namespace Mapping for Filesystem (#6009)

Commit:
Added AuthorizationEnforcementModule. Moved RemotePrivilegesFetcher to cdap-security and RemoteOpsClient/MethodArgument to cdap-common so they can be accessed by both cdap-app-fabric and cdap-data-fabric.

Commit:
CDAP-6321: Fixed checkstyle errors and added default values for binding ports

anew commented:
Not sure what happened, @bennythomps, did you really add 100+ commits?

bennythomps commented:
Messed up my fork....rebased and reopening
